public/video-ui/src/components/VideoData/VideoData.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import {ManagedForm, ManagedField, ManagedSection} from '../ManagedForm';
+import TextInput from '../FormFields/TextInput';
+import TextArea from '../FormFields/TextArea';
+import SelectBox from '../FormFields/SelectBox';
+import CheckBox from '../FormFields/CheckBox';
+import DatePicker from '../FormFields/DatePicker';
+import KeywordPicker from '../FormFields/KeywordPicker';
+import {fieldLengths} from '../../constants/videoEditValidation';
+import {videoCategories} from '../../constants/videoCategories';
+import {privacyStates} from '../../constants/privacyStates';
+
+class VideoData extends React.Component {
+
+  hasCategories = () => this.props.youtube.categories.length !== 0;
+  hasChannels = () => this.props.youtube.channels.length !== 0;
+
+  componentWillMount() {
+    if (! this.hasCategories()) {
```

Comment:
I think its done because if people navigate around the app without refreshing the page the youtube channels and categories will remain stored in the properties so there is no need to fetch them again.